contrib/kktix_bq_etl.py

```diff
@@ -0,0 +1,559 @@
+
+    parser.add_argument(
+        "-d", "--dataset-name", help="BigQuery dataset name to create or append"
+    )
+
+    parser.add_argument(
+        "-t", "--table-name", help="BigQuery table name to create or append"
+    )
+
+    parser.add_argument(
+        "-k", "--ticket-type", help="Attendee ticket-type name in corporate, individual, reserved"
+    )    
+
+    parser.add_argument(
+        "-y", "--ticket-year", help="PyConTW year"
+    )    
+
+
+    parser.add_argument(
+        "--upload",
+        action="store_true",
+        help="Parsing the file but not upload it",
+        default=False,
+    )
+
+    args = parser.parse_args()
+
+    if args.project_id: 
+        project_id = args.project_id
+    if args.dataset_name:             
+        dataset_id = args.dataset_name
+    if args.table_name:    
+        table_id = args.table_name
+    if args.ticket_type:    
+        ticket_type = args.ticket_type        
+    if args.ticket_year:    
+        ticket_year = args.ticket_year            
+
+    # Set up the client object
+    client = bigquery.Client(project=project_id)
+
+    # Set up the job config
+    job_config = bigquery.QueryJobConfig()
+    job_config.use_legacy_sql = False
+
+    # Build the SQL query to extract the data
+    query = f'SELECT * FROM `{TABLE}` WHERE lower( NAME ) LIKE "%{ticket_year}%{ticket_type}%"'
+
+    # Execute the query and extract the data 
+    # bigquery depends on db_dtypes for to_dataframe(), depends on apache-arrow, cannot be installed (build failed) on my old MAC
+    # use the raw result from bigQuery is good for this use case
+    job = client.query(query, job_config=job_config)
+    results = job.result() 
+
+    # print(results)
+
+    print(f'Extracted from {TABLE}  successful.')
+  
+    # Use DataFrame for table transform operations
+    df = pd.DataFrame()
+    for row in results:
+        json_dict = json.loads(row['ATTENDEE_INFO'])
+        df_dict = pd.DataFrame([json_dict])
+
+        # We don't have paid_date column from ATTENDEE_INFO, convert the updated_at timestamp
+        timestamp = df_dict.loc[0, 'updated_at']
+        # print(timestamp)
+        dt_object = datetime.fromtimestamp(timestamp)
+        transc_date = dt_object.strftime('%Y-%m-%d')
+        # print(transc_date)    
+        df_dict['paid_date'] = transc_date
+
+        # ticket_type value is always "qrcode" here, and conflict with old table, 
+        # kyc, id_number, slot value is always empty
+        # 'id','ticket_id','state','checkin_code', 'qrcode', updated_at, order_no no exist in old table
+        # data will be handle later
+        useless_columns = ['id','ticket_id','state','checkin_code', 'qrcode', 'currency', 'data', 'ticket_type', 'kyc', 'updated_at', 'id_number', 'slot', 'order_no']
+        df_dict = df_dict.drop(columns = useless_columns) 
+        df_dict = df_dict.rename(columns = {"reg_no": "registration_no", "ticket_name": "ticket_type", "is_paid": "payment_status"})
```

Comment:
what do you thiink if we keep all of the original data in the warehouse, and only deserialize/extract those "useful" columns from JSON to another DB layer?

Reply:
Yes, we should keep all the of the original data that maybe useful someday.